### ISSUE TITLE
Jest tests won't complete properly when debugging

### ISSUE THREAD

**Issue by markfinlabs** [microsoft/vscode-js-debug#222]
Hi there,
I gave the new debugger a quick try with my Jest tests to try parallel testing (no --runInBand) having noticed https://github.com/microsoft/vscode-js-debug/issues/214.

First thing I noticed is compared to a parallel run without the debugger speed seems very variable. Sometimes the tests don't seem to finish at all.... other times they take just a bit longer than a non-debug run.

Also even after a successful run Jest doesn't close properly - last message on terminal is as below and still-active parent process needs to be stopped to finish debugging.

```
Jest did not exit one second after the test run has completed.
This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

I can't share my project but tests use Puppeteer (automated web testing) so are quite heavy, maybe that is an issue? On the other hand, running tests in parallel without the debugger works fine on the command line.

My config is:

```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug Tests (All)",
      "type": "node",
      "request": "launch",
      //"trace": true,
      "runtimeArgs": [
        //"--inspect-brk",
        "${workspaceRoot}/node_modules/.bin/jest",
        //"--runInBand",
        "--updateSnapshot",
      ],
      "console": "integratedTerminal",
      "internalConsoleOptions": "neverOpen",
      "port": 9229
    }
  ]
}
```

**Comment by connor4312** [comment]
Thanks for opening an issue!

Some slowness which affects Jest should be fixed in tomorrow's Insiders build.

Next time you run into an issue with it not shutting down, could you enable that `trace: true` flag and share the logs with me?

 > **⚠️Warning:** The logs will include information from your workspace, such as file paths. If you have any concerns about posting this publicly on Github, you can alternatively email them to me at connor@xbox.com. On our side, we only use these logs to investigate issues like this.

**Comment by markfinlabs** [comment]
Hi Connor, just to say thanks for the reply and sorry for slow response from me. Things are better now although there seems to be still a (slightly different) termination issue.

However I unfortunately have updated Jest and macOS (to Catalina) as well as getting debugger updates. So let me experiment over the next week and post an update here.... I'm really only interested in remaining problems when using the latest of everything.

**Comment by connor4312** [comment]
That's fine, I am as well 🙂 

If you do get reproductions of the issue, please share logs here. If it's resolved for you in latest versions, feel free to close this.

**Comment by markfinlabs** [comment]
Sure, will do!

**Comment by markfinlabs** [comment]
Just to check in to say I'm still testing and sometimes seeing error below at end of run. I'll keep trying different things and of course just received updates to VS Code and the debugger.

illegal access
FATAL ERROR: 
node::inspector::Agent::ToggleAsyncHook Cannot toggle Inspector's AsyncHook, please report this.
 1: 0x100ba1aee node::Abort() (.cold.1) [/usr/local/bin/node]
 2: 0x100084b65 node::FatalError(char const*, char const*) [/usr/local/bin/node]
 3: 0x100084c8d node::OnFatalError(char const*, char const*) [/usr/local/bin/node]
 4: 0x100084b6e node::OnFatalError(char const*, char const*) [/usr/local/bin/node]
 5: 0x10011fe1d node::inspector::Agent::ToggleAsyncHook(v8::Isolate*, v8::Global<v8::Function> const&) [/usr/local/bin/node]
 6: 0x1005ffb39 v8_inspector::V8Debugger::setAsyncCallStackDepth(v8_inspector::V8DebuggerAgentImpl*, int) [/usr/local/bin/node]
 7: 0x10060423d v8_inspector::V8DebuggerAgentImpl::disable() [/usr/local/bin/node]
 8: 0x100612d6c v8_inspector::V8InspectorSessionImpl::~V8InspectorSessionImpl() [/usr/local/bin/node]
 9: 0x100612f7c v8_inspector::V8InspectorSessionImpl::~V8InspectorSessionImpl() [/usr/local/bin/node]
10: 0x100120c6b node::inspector::(anonymous namespace)::ChannelImpl::~ChannelImpl() [/usr/local/bin/node]
11: 0x100120cda node::inspector::(anonymous namespace)::ChannelImpl::~ChannelImpl() [/usr/local/bin/node]
12: 0x100120e98 std::__1::unique_ptr<std::__1::__hash_node<std::__1::__hash_value_type<int, std::__1::unique_ptr<node::inspector::(anonymous namespace)::ChannelImpl, std::__1::default_delete<node::inspector::(anonymous namespace)::ChannelImpl> > >, void*>, std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<int, std::__1::unique_ptr<node::inspector::(anonymous namespace)::ChannelImpl, std::__1::default_delete<node::inspector::(anonymous namespace)::ChannelImpl> > >, void*> > > >::~unique_ptr() [/usr/local/bin/node]
13: 0x100121256 node::inspector::NodeInspectorClient::disconnectFrontend(int) [/usr/local/bin/node]
14: 0x100121013 node::inspector::(anonymous namespace)::SameThreadInspectorSession::~SameThreadInspectorSession() [/usr/local/bin/node]
15: 0x10012105c node::inspector::(anonymous namespace)::SameThreadInspectorSession::~SameThreadInspectorSession() [/usr/local/bin/node]
16: 0x10012efd4 std::__1::unique_ptr<node::inspector::(anonymous namespace)::MainThreadSessionState, std::__1::default_delete<node::inspector::(anonymous namespace)::MainThreadSessionState> >::~unique_ptr() [/usr/local/bin/node]
17: 0x10012f01e node::inspector::(anonymous namespace)::DeletableWrapper<node::inspector::(anonymous namespace)::MainThreadSessionState>::~DeletableWrapper() [/usr/local/bin/node]
18: 0x10012f814 std::__1::unique_ptr<std::__1::__hash_node<std::__1::__hash_value_type<int, std::__1::unique_ptr<node::inspector::Deletable, std::__1::default_delete<node::inspector::Deletable> > >, void*>, std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<int, std::__1::unique_ptr<node::inspector::Deletable, std::__1::default_delete<node::inspector::Deletable> > >, void*> > > >::~unique_ptr() [/usr/local/bin/node]
19: 0x10012f85c unsigned long std::__1::__hash_table<std::__1::__hash_value_type<int, std::__1::unique_ptr<node::inspector::Deletable, std::__1::default_delete<node::inspector::Deletable> > >, std::__1::__unordered_map_hasher<int, std::__1::__hash_value_type<int, std::__1::unique_ptr<node::inspector::Deletable, std::__1::default_delete<node::inspector::Deletable> > >, std::__1::hash<int>, true>, std::__1::__unordered_map_equal<int, std::__1::__hash_value_type<int, std::__1::unique_ptr<node::inspector::Deletable, std::__1::default_delete<node::inspector::Deletable> > >, std::__1::equal_to<int>, true>, std::__1::allocator<std::__1::__hash_value_type<int, std::__1::unique_ptr<node::inspector::Deletable, std::__1::default_delete<node::inspector::Deletable> > > > >::__erase_unique<int>(int const&) [/usr/local/bin/node]
20: 0x10012e573 node::inspector::MainThreadInterface::RemoveObject(int) [/usr/local/bin/node]
21: 0x10012e14c node::inspector::MainThreadInterface::DispatchMessages() [/usr/local/bin/node]
22: 0x10012f504 node::inspector::(anonymous namespace)::DispatchMessagesTask::Run() [/usr/local/bin/node]
23: 0x1000d493d node::PerIsolatePlatformData::RunForegroundTask(std::__1::unique_ptr<v8::Task, std::__1::default_delete<v8::Task> >) [/usr/local/bin/node]
24: 0x1000d3d04 node::PerIsolatePlatformData::FlushForegroundTasksInternal() [/usr/local/bin/node]
25: 0x1006c2e10 uv__async_io [/usr/local/bin/node]
26: 0x1006d2650 uv__io_poll [/usr/local/bin/node]
27: 0x1006c324a uv_run [/usr/local/bin/node]
28: 0x10003ea04 node::Environment::CleanupHandles() [/usr/local/bin/node]
29: 0x10003ed71 node::Environment::RunCleanup() [/usr/local/bin/node]
30: 0x1000b67f0 node::NodeMainInstance::Run() [/usr/local/bin/node]
31: 0x100061c06 node::Start(int, char**) [/usr/local/bin/node]
32: 0x7fff683267fd start [/usr/lib/system/libdyld.dylib]

**Comment by connor4312** [comment]
Well, that's a Node.js bug! It looks like there've been a few issues on their repo with this error in the past. What's the exact Node.js version you're running?

**Comment by markfinlabs** [comment]
Interesting.

```
node --version
v13.7.0
```

**Comment by connor4312** [comment]
Okay, thanks. I brought this up in an issue under Node.js. Closing this here as it'll be an upstream fix.

**Comment by markfinlabs** [comment]
Thanks and no problem, I'll keep observing things and open a new issue for anything else I see independent of the Node.js issue.
